src/Http/Authentication.Core/src/AuthenticationMetrics.cs

```diff
@@ -0,0 +1,248 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Diagnostics.Metrics;
+using System.Runtime.CompilerServices;
+using Microsoft.AspNetCore.Http;
+
+namespace Microsoft.AspNetCore.Authentication;
+
+internal sealed class AuthenticationMetrics
+{
+    public const string MeterName = "Microsoft.AspNetCore.Authentication";
+
+    private readonly Meter _meter;
+    private readonly Histogram<double> _authenticatedRequestDuration;
+    private readonly Counter<long> _challengeCount;
+    private readonly Counter<long> _forbidCount;
+    private readonly Counter<long> _signInCount;
+    private readonly Counter<long> _signOutCount;
+
+    public AuthenticationMetrics(IMeterFactory meterFactory)
+    {
+        _meter = meterFactory.Create(MeterName);
+
+        _authenticatedRequestDuration = _meter.CreateHistogram<double>(
+            "aspnetcore.authentication.authenticate.duration",
+            unit: "s",
+            description: "The authentication duration for a request.",
+            advice: new() { HistogramBucketBoundaries = MetricsConstants.ShortSecondsBucketBoundaries });
+
+        _challengeCount = _meter.CreateCounter<long>(
+            "aspnetcore.authentication.challenges",
+            unit: "{request}",
+            description: "The total number of times a scheme is challenged.");
+
+        _forbidCount = _meter.CreateCounter<long>(
+            "aspnetcore.authentication.forbids",
+            unit: "{request}",
+            description: "The total number of times an authenticated user attempts to access a resource they are not permitted to access.");
+
+        _signInCount = _meter.CreateCounter<long>(
+            "aspnetcore.authentication.sign_ins",
+            unit: "{request}",
+            description: "The total number of times a principal is signed in.");
+
+        _signOutCount = _meter.CreateCounter<long>(
+            "aspnetcore.authentication.sign_outs",
+            unit: "{request}",
+            description: "The total number of times a scheme is signed out.");
+    }
+
+    public void AuthenticatedRequestSucceeded(string? scheme, AuthenticateResult result, long startTimestamp, long currentTimestamp)
```

Comment:
Nit: This includes failed `AuthenticateResult` which was harder to understand at a glance with "Succeeded" in the method name. Could we combine this with `AuthenticatedRequestFailed` and make the `AuthenticateResult` and `Exception` both nullable. I think we should include the `AuthenticateResult.Failure` exception type in the error tag too.

```suggestion
    public void AuthenticatedRequestCompleted(string? scheme, AuthenticateResult result, long startTimestamp, long currentTimestamp)
```

Reply:
Yeah, I agree the "succeeded" terminology is confusing. The `Succeeded` and `Failed` naming probably should have been `Completed` and `UnhandledException`, so the possible outcomes would be structured like:
* Completed
  * Succeeded
  * Failed
* Unhandled exception

We could include the failure exception in `error.type`, but that does change the OTel API a bit (rather than there being either an `authentication.result` or `error.kind`, there's a possibility of both). But I don't have an objection to making that change. It _would_ however make things a little inconsistent with authorization, since an authorization failure doesn't necessarily imply an exception (so the `error.kind` would be empty in the failure case). Not sure if that's really a big deal though.

Reply:
> It would however make things a little inconsistent with authorization, since an authorization failure doesn't necessarily imply an exception (so the error.kind would be empty in the failure case). Not sure if that's really a big deal though.

I'm personally fine with this inconsistency. The authn and authz APIs aren't exactly symmetrical, and this is an artifact of that.